### PR TITLE
Specs for endless and beginless for ArithemticSequence

### PR DESCRIPTION
**Pull request by edipofederle**

### CODE REVIEW

**Comment by edipofederle** [comment]
This is relevant? If so, any tip to fix the rubocop? thx!

File: core/enumerator/arithmetic_sequence/beginless_spec.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require_relative '../../../spec_helper'
+
+describe "Enumerator::ArithmeticSequence beginless" do
+  it "returns an instance of Enumerator::ArithmeticSequence" do
+    eval("(..10).step(1)").should be_an_instance_of(Enumerator::ArithmeticSequence)
+    eval("(...10).step(1)").should be_an_instance_of(Enumerator::ArithmeticSequence)
+  end
+end
```
Comment (eregon) [inline core/enumerator/arithmetic_sequence/beginless_spec.rb:8]: This is already tested in https://github.com/oracle/truffleruby/blob/baa94b35af7fca1fcf87981009c4f5896de046d6/spec/ruby/core/range/step_spec.rb#L477-L516 and it'll get in this repo soon.

File: core/enumerator/arithmetic_sequence/endless_spec.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require_relative '../../../spec_helper'
+
+describe "Enumerator::ArithmeticSequence endless" do
+  it "returns an instance of Enumerator::ArithmeticSequence" do
+    (10..).step(1).should be_an_instance_of(Enumerator::ArithmeticSequence)
+    (10..).step(1).should be_an_instance_of(Enumerator::ArithmeticSequence)
+  end
+end
```
Comment (eregon) [inline core/enumerator/arithmetic_sequence/endless_spec.rb:8]: Same, this is already tested in https://github.com/oracle/truffleruby/blob/baa94b35af7fca1fcf87981009c4f5896de046d6/spec/ruby/core/range/step_spec.rb#L477-L516

**Review by eregon — APPROVED** [review]
Thanks, I think it's good to check what begin/end return for those cases.

File: core/enumerator/arithmetic_sequence/beginless_spec.rb (hunk 1 above)
Comment (eregon) [reply core/enumerator/arithmetic_sequence/beginless_spec.rb:8]: Could you remove this and core/enumerator/arithmetic_sequence/endless_spec.rb ?

Comment (edipofederle) [reply core/enumerator/arithmetic_sequence/beginless_spec.rb]: done

File: core/enumerator/arithmetic_sequence/endless_spec.rb (hunk 2 above)
Comment (edipofederle) [reply core/enumerator/arithmetic_sequence/endless_spec.rb]: removed

File: core/enumerator/arithmetic_sequence/begin_spec.rb [hunk 3]
```diff
@@ -6,4 +6,11 @@
     (1..10).step.begin.should == 1
     (1...10).step.begin.should == 1
   end
+
+  context "with beginless" do
```
Comment (eregon) [inline core/enumerator/arithmetic_sequence/begin_spec.rb:10]: This needs a version guard like `ruby_version_is "2.7" do`

**Pull request merged**